tests/PageCollector.test.ts

```diff
@@ -250,4 +250,40 @@ describe('NetworkCollector', () => {
     assert.equal(collector.getData(page)[0], navRequest);
     assert.equal(collector.getData(page)[1], request2);
   });
+
+  it('correctly picks up after multiple back to back navigations', async () => {
```

Comment:
let's add a e2e test that involves actually formatting requests from multiple navigations

Reply:
It's not about this specific test: a general comment for better test coverage. This test covers the PageCollector but does not actually verify that we can format requests from earlier navigations successfully. E.g., bodies could be not available anymore and we need to make sure we handle that.